rfcs/proposed/task_group_wait_single_task/README.md

```diff
@@ -0,0 +1,496 @@
+
+Returns ``task_status::completed`` if the task was executed, or ``task_status::canceled`` if it was not.
+
+### Member functions of ``task_arena`` class
+
+```cpp
+task_status wait_for(task_completion_handle& comp_handle);
+```
+
+Waits for the completion of the task represented by ``comp_handle``.
+If completion was transferred to another task using ``tbb::task_group::transfer_completion_to``, the function waits for completion of that task.
+
+This is semantically equivalent to: ``execute([&] { tg.wait_for(comp_handle); })``.
+
+Returns: ``task_status::completed`` if the task was executed and ``task_status::canceled`` otherwise.
+
+Introducing ``wait_for`` in the ``this_task_arena`` namespace is unnecessary, as it would be functionally identical to calling the ``task_group::wait_for`` API directly.
+
+## Future Enhancements
+
+### Run-and-wait methods for ``task_arena``
+
+It may be beneficial to introduce APIs that simplify task submission into ``task_arena`` and support waiting for task completion.
+
+This functionality would be semantically similar to:
+
+```cpp
+tbb::task_group tg;
+tbb::task_arena arena;
+tbb::task_handle task = tg.defer(...);
+
+tbb::task_completion_handle comp_handle = task;
+arena.enqueue(std::move(task));
+task_status status = arena.wait_for(comp_handle);
+```
+
+One of possible name for such an extension is ``task_arena::enqueue_and_wait_for(task_handle&& task)``.
+This function is semantically close to ``task_arena::execute`` (which may submit the task and waits for its completion), but it guarantees that
+the task is enqueued into the arena.
+
+A similar function could also be added to the ``tbb::this_task_arena`` namespace.
+
+### Work Isolation Issue
+
+Similar to other ``task_arena::wait_for`` overload that take a ``task_group`` argument, it makes sense to also consider work isolation for
+the APIs introduced in this proposal.
+If the wait is unrestricted, the waiting thread may execute any task from the arena, including those unrelated to the completion of the awaited
+task or its associated ``task_group``.
+
+Ideally, the waiting thread should only execute tasks that contribute to the completion of the awaited task. In such a model, all tasks
+within the same subgraph would need to share a common isolation tag. In theory, a successor could inherit the isolation tag from
+its predecessor. However, multiple predecessors will have different tags.
+
+This would require changes to the current isolation mechanism to support multiple isolation tags, allowing a thread waiting for a successor
+to execute tasks with any of its predecessor's tags. 
+
+As an initial step, it makes sense to isolate the waiting thread so that it only executes tasks related to the same ``task_group`` as the awaited task - similar to the improvement described in the [RFC for another overload of ``task_arena::wait_for``](../task_arena_waiting/task_group_interop.md).
+
+## Exit Criteria and Open Questions
```

Comment:
Is the intention to go through an experimental phase, or straight to production?